src/main/kotlin/no/template/controller/VersionApiImpl.kt

```diff
@@ -0,0 +1,32 @@
+package no.template.controller
+
+import no.template.generated.model.Version
+import javax.servlet.http.HttpServletRequest
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
+import java.util.*
+
+
+@Controller
+open class VersionApiImpl : no.template.generated.api.VersionApi {
+
+    private var properties : Properties? = null
+
+
+    override fun getVersion(httpServletRequest: HttpServletRequest?): ResponseEntity<Version> {
+        if (properties == null) {
+            properties = Properties()
+            properties!!.load(this::class.java.getResourceAsStream("/git.properties"))
+        }
+
+        val version = Version()
+        version.branchName = properties!!.getProperty("git.branch")
```

Comment:
Bruk ? i stedet for !!.
Med ? settes branchName til null om properties er null, med !! får vi nullpointer